### ISSUE TITLE
Featured Image block uses post title as ALT text instead of media attribute when linked

### ISSUE THREAD

**Issue by galbaras** [WordPress/gutenberg#70723]
### Description

In a query loop, when the featured image setting "Link to post" is enabled, the ALT text of that image is the post title, rather than the ALT text of the image from the media library. This is incorrect, given that the ALT text is meant to describe the image content, not the link target.

This is deliberately done in the code. See [here](https://github.com/WordPress/gutenberg/blob/795734c9741904eb00243bae1f3d787512052e03/packages/block-library/src/post-featured-image/index.php#L29).

### Step-by-step reproduction instructions

1. Pick a post X
2. Set its featured image to image Y
3. Set the ALT text attribute of Y in the media library to be different from the title of X
4. Edit an archive template
5. Select the Featured Image block in the Query Loop
6. Enable "Link to post"
7. Save the template
8. View an archive page containing X
9. Check the ALT text of Y

### Screenshots, screen recording, code snippet

"Link to post" disabled:

<img width="847" height="420" alt="Image" src="https://github.com/user-attachments/assets/8020a64c-b0e2-4040-a340-afd1985cbc05" />

"Link to post" enabled:

<img width="1138" height="404" alt="Image" src="https://github.com/user-attachments/assets/993f61ee-df00-485c-9398-ce94c1bea246" />

### Environment info

WordPress 6.8.1
Child theme of Twenty Twenty-Five

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
Thanks for the report.

> when the featured image setting "Link to post" is enabled, the ALT text of that image is the post title, rather than the ALT text of the image from the media library. This is incorrect, given that the ALT text is meant to describe the image content, not the link target.

We need to explore whether this is really incorrect, since the current implementation is based on exactly the opposite reasoning. The current implementation is intentional, and you can see why in #39634:

> The problem here, is that `wp_get_attachment_image` uses the image's metadata to get the alt-text. As a result of that, assistive technologies announce the image's alt text (if one even exists) for the link, and that is not what it should be.
The link points to the post, and therefore should have the post-title as alt description, instead of the image's alt description. The image does not describe the link, the post-title does.

Personally, I think the current implementation is correct.

cc @aristath @Mamaduka @alexstine

**Comment by Mamaduka** [comment]
I'll defer to the A11y team on this one, but it's also my understanding that the current behavior is correct.

cc @joedolson

**Comment by galbaras** [comment]
Thank you for the explanation, but it seems to me that the post title should then be used in the link's `title` or `aria-label` attribute, rather than the image's ALT text. Another option is to add a hidden assistive text SPAN into the link.

**Comment by t-hamano** [comment]
I think just the images' ALT text should be enough. NVDA (Windows Screen Reader) speaks `figure {post_title} link` when the linked featred image is focused.

See: https://www.w3.org/WAI/tutorials/images/functional/#example-1-image-used-alone-as-a-linked-logo

**Comment by galbaras** [comment]
ALT text is used to enhance accessibility, but also SEO, so I wonder how Google views an image with possibly-irrelevant text.

What does NVDA say when there's a `title` or `aria-label` on the link, as well as ALT text on the image?

**Comment by t-hamano** [comment]
I checked how NVDA reads out when the link is focused with the following HTML:

```html
<figure class="wp-block-post-featured-image">
  <a href="http://example.com" aria-label="Aria Label">
    <img src="https://placehold.co/600x400/EEE/31343C" alt="Alt Text">
  </a>
</figure>
```

Only the `aria-label` value is spoken, not the `alt` text:

```
figure
Aria Label  link
```

I'm not sure if this HTML is appropriate.

**Comment by galbaras** [comment]
Seems perfect to me. The screen reader "sees" the aria label, and Google sees the ALT text. Everybody wins 😄

**Comment by anevins12** [comment]
Hi @galbaras,

Assistive technology users, such as screen reader users, need to know where the link leads to. 

Can you go into more detail as to why screen reader users should be receiving the description of the image, rather than the destination of the link?

It sounds like the underlying issue we're leading to is that the image _link_ is redundant, should be removed, and the image can then be described as intended by the OP.

**Comment by joedolson** [comment]
Alt text is not intended to describe the image; it is intended to describe the purpose and meaning of the image. If the image is a link, then the purpose of the image is to direct the user to another location, and the alt text needs to communicate where the user will go. The image itself is largely irrelevant in that case.

Ideally, it would be possible for individual users to override that alt text, but it would always need to *contain* the post title, as that would continue to be the purpose of the link.

Take a look at the [W3C's alt text decision tree](https://www.w3.org/WAI/tutorials/images/decision-tree/) (which is also linked in the editor when setting alt text for an image).

There are edge cases: for example, if the image is a link and *also* contains text, then the link text should be the image text. However, this isn't a case that can be determined programmatically.

**Comment by galbaras** [comment]
Can't argue with the standard, I guess. Thank you for discussing.

**Comment by Mamaduka** [comment]
Thanks for the feedback, everyone!
